src/Data/Nat/Show.agda

```diff
@@ -8,17 +8,78 @@
 
 module Data.Nat.Show where
 
-open import Data.Nat
+open import Data.Char as Char using (Char)
+open import Data.Digit using (showDigit; toDigits)
+open import Function   using (_∘_; _$_)
+open import Data.List  using (List; []; _∷_; map; reverse)
+open import Data.Nat   using (ℕ; _≟_; suc; pred; _+_; _*_; _<_; _>_; z≤n; s≤s;
+                                                                      _≤?_)
+open import Data.Nat.DivMod     using (_div_; _%_; a≡a%n+[a/n]*n; [a/n]*n≤a)
+open import Data.Product        using (proj₁)
+open import Data.Nat.Properties using
+                 (+-comm; *-identityʳ; <-transˡ; <-irrefl; <⇒≢; n≢0⇒n>0;
+                  m≤m+n; *-monoʳ-<; m<m*n; m≢0⇒suc[pred[m]]≡m
+                 )
+open import Data.String as String using (String)
+open import Induction.Nat         using (<-wellFounded)
+open import Induction.WellFounded using (Acc; acc)
+open import Relation.Binary.PropositionalEquality using (_≡_; _≢_; refl; sym;
+                                                         cong; ≢-sym)
+open import Relation.Nullary           using (yes; no)
 open import Relation.Nullary.Decidable using (True)
-open import Data.String.Base as String using (String)
-open import Data.Digit
-open import Data.Product using (proj₁)
-open import Function
-open import Data.List.Base
+open import Relation.Nullary.Negation  using (contradiction)
 
--- showInBase b n is a string containing the representation of n in
--- base b.
 
+------------------------------------------------------------------------------
+-- Conversion from unary representation to the representation by the given
+-- base.
+toDigitNats : (base : ℕ) → base > 1 → ℕ → List ℕ
+toDigitNats 0             ()
+toDigitNats 1             (s≤s ())
+toDigitNats (suc (suc b)) base>1 x =  aux x (<-wellFounded x) []
+  where
+  -- <-wellFounded  means that _<_ is well-founded on ℕ
+  --                ("each number in ℕ is accessible from 0 by _<_").
+
+  base = suc (suc b);  pbase = pred base
+
+  aux : (n : ℕ) → Acc _<_ n → List ℕ → List ℕ
+  aux 0       _        =  (0 ∷_)
+  aux (suc n) (acc wf) =  aux'
+    where
+    n' = suc n;   q = n' div base;   r = n' % base;   pq = pred q
+
+    n'≡r+q*base :  n' ≡ r + q * base
+    n'≡r+q*base =  a≡a%n+[a/n]*n n' pbase
+
+    aux' : List ℕ → List ℕ
+    aux' with q ≟ 0
+    ... | yes _   =  (r ∷_)
+    ... | no q≢0 =  aux q (wf _ q<n') ∘ (r ∷_)   -- use  n' ≡ r + q*base
+      where
+      q>0       = n≢0⇒n>0 q≢0
```

Comment:
This line can also be removed if instead of checking `q ≟ 0` in the line above, you instead check `0 <? q` and then invert the `yes` and `no` clauses.